visualization/app/codeCharta/codeCharta.service.spec.ts

```diff
@@ -154,39 +158,85 @@ describe("codeChartaService", () => {
 			expect(CodeChartaService.ROOT_PATH).toEqual(`/${expected.map.name}`)
 		})
 
-		it("should not replace the last files when loading new files", () => {
-			codeChartaService.loadFiles([
-				{
-					fileName: "FirstFile",
-					content: validFileContent,
-					fileSize: 42
-				}
+		it("should replace files with equal file name and checksum when loading new files", async () => {
+			const validContent2ndFile = klona(validFileContent)
+			validContent2ndFile.fileChecksum = "validHash_1"
+
+			await codeChartaService.loadFiles([{ fileName: "FirstFile", content: validFileContent, fileSize: 42 }])
+
+			await codeChartaService.loadFiles([
+				{ fileName: "FirstFile", content: validFileContent, fileSize: 42 },
+				{ fileName: "SecondFile", content: validContent2ndFile, fileSize: 42 }
+			])
+
+			expect(getCCFiles(storeService.getState().files).length).toEqual(2)
+			expect(getCCFiles(storeService.getState().files)[0].fileMeta.fileName).toEqual("FirstFile")
+			expect(getCCFiles(storeService.getState().files)[0].fileMeta.fileChecksum).toEqual("invalid-md5-sample")
+			expect(getCCFiles(storeService.getState().files)[1].fileMeta.fileName).toEqual("SecondFile")
+			expect(getCCFiles(storeService.getState().files)[1].fileMeta.fileChecksum).toEqual("validHash_1")
+		})
+
+		it("should load files with equal file name but different checksum and rename uploaded files ", async () => {
+			const validContent2ndFile = klona(validFileContent)
+			validContent2ndFile.fileChecksum = "validHash_1"
+			const validContent3ndFile = klona(validFileContent)
+			validContent3ndFile.fileChecksum = "validHash_2"
+
+			await codeChartaService.loadFiles([{ fileName: "FirstFile", content: validFileContent, fileSize: 42 }])
+
+			await codeChartaService.loadFiles([
+				{ fileName: "FirstFile", content: validContent2ndFile, fileSize: 42 },
+				{ fileName: "FirstFile", content: validContent3ndFile, fileSize: 42 }
+			])
+
+			expect(getCCFiles(storeService.getState().files).length).toEqual(3)
```

Comment:
Minor note: It would be a little shorter to store the files in a variable instead of getting all CC-Files again and again :-)